src/sentry/models/notificationaction.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import annotations
+
+import logging
+from collections import defaultdict
+from enum import IntEnum
+from typing import Callable, Iterable, List, Tuple
+
+from django.db import models
+from pyparsing import MutableMapping
+
+from sentry.db.models import FlexibleForeignKey, Model, sane_repr
+from sentry.db.models.base import region_silo_only_model
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+
+logger = logging.getLogger(__name__)
+
+
+class FlexibleIntEnum(IntEnum):
+    @classmethod
+    def as_choices(cls) -> Iterable[Tuple[int, str]]:
+        raise NotImplementedError
+
+    @classmethod
+    def get_name(cls, value: int) -> str | None:
+        return dict(cls.as_choices()).get(value)
+
+    @classmethod
+    def get_value(cls, name: str) -> int | None:
+        invert_choices = {v: k for k, v in dict(cls.as_choices()).items()}
+        return invert_choices.get(name)
+
+
+class ActionServiceType(FlexibleIntEnum):
+    """
+    The available services to fire action notifications
+    """
+
+    EMAIL = 0
+    PAGERDUTY = 1
+    SLACK = 2
+    MSTEAMS = 3
+    SENTRY_APP = 4
+    SENTRY = 5  # Use notification platform (src/sentry/notifications)
+
+    @classmethod
+    def as_choices(cls) -> Iterable[Tuple[int, str]]:
+        return (
+            (cls.EMAIL, "email"),
+            (cls.PAGERDUTY, "pagerduty"),
+            (cls.SLACK, "slack"),
+            (cls.MSTEAMS, "msteams"),
+            (cls.SENTRY_APP, "sentry_app"),
+            (cls.SENTRY, "sentry"),
```

Comment:
Should we call this `sentry notifications` :shrug: having sentry + sentryapp could be confusing for future us.